core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -179,6 +179,10 @@ private TableProperties() {
   // If not set, defaults to a "metadata" folder underneath the root path of the table.
   public static final String WRITE_METADATA_LOCATION = "write.metadata.path";
 
+  public static final String PINTEREST_WRITE_METADATA_PREFIX = "pinterest.write.metadata.prefix";
```

Comment:
Why does it have to be pinterest specific prop? Let's keep it generic and open a PR against OSS instead?

Reply:
Also I think we should use `write.metadata.path.prefix` instead.